src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Error.cs

```diff
@@ -93,13 +92,13 @@ public override bool CheckTypes(CheckTypesContext context, TexlNode[] args, DTyp
                 return false;
             }
 
-            var argumentKindType = names.First(tn => tn.Name == requiredKindField.Name).Type;
+            var argumentKindType = names.FirstOrDefault(tn => tn.Name == requiredKindField.Name).Type;
```

Comment:
>FirstOrDefault [](http://example.com/codeflow?start=41&length=14)

Calling `FirstOrDefault` implies that `null` can be returned, but the code is dereferencing it right after it (`.Type`). We know that this will not return null, since this case is handled by the previous `if` statement, so leaving it as `First` is fine.

Reply:
updated code. thanks